java/mapreduce/src/java/org/apache/orc/mapred/OrcMapredRecordReader.java

```diff
@@ -98,16 +101,22 @@ public boolean next(NullWritable key, V value) throws IOException {
     if (!ensureBatch()) {
       return false;
     }
+    int rowIdx = batch.selectedInUse ? batch.selected[rowInBatch] : rowInBatch;
     if (schema.getCategory() == TypeDescription.Category.STRUCT) {
       OrcStruct result = (OrcStruct) value;
       List<TypeDescription> children = schema.getChildren();
       int numberOfChildren = children.size();
       for(int i=0; i < numberOfChildren; ++i) {
-        result.setFieldValue(i, nextValue(batch.cols[i], rowInBatch,
-            children.get(i), result.getFieldValue(i)));
+        TypeDescription child = children.get(i);
```

Comment:
unrelated change?

Reply:
Sorry, yes this should have been a separate bug fix patch that MapRed reader does not respect inclusions when setting the values.

Will pull this out.